### PR TITLE
Hookup "view desktop" link to delete cookie and reload

### LINKED ISSUE #1341: Unset "mamo" cookie and refresh page from "View desktop site" link
This will make more sense once the mobile pages are served via the same host as AMO. 

The "View desktop site" link in the footer should remove the mamo cookie and refresh the page as per the existing mobile page functionality.

See also https://github.com/mozilla/addons-server/issues/3712

### PR DESCRIPTION
**Pull request by muffinresearch**
Fixes #1341

### CODE REVIEW

File: src/amo/components/Footer.js [hunk 1]
```diff
@@ -12,10 +12,11 @@ export class FooterBase extends React.Component {
   static propTypes = {
     i18n: PropTypes.object.isRequired,
     lang: PropTypes.string.isRequired,
+    handleViewDesktop: PropTypes.func.isRequired,
```
Comment (tofumatt) [inline src/amo/components/Footer.js]: OCD in me would love if this were alphabetised ^_^

File: src/amo/containers/App.js [hunk 2]
```diff
@@ -30,6 +31,14 @@ export class AppBase extends React.Component {
     MastHeadComponent: MastHead,
   }
 
+  onViewDesktop = (event, { window_ = window, cookie_ = cookie } = {}) => {
+    event.preventDefault();
+    if (window_ && window_.location) {
+      cookie_.remove('mamo', { path: '/' });
+      window_.location.reload();
```
Comment (tofumatt) [inline src/amo/containers/App.js]: http://tinyurl.com/5pc8k4

File: tests/client/amo/containers/TestApp.js [hunk 3]
```diff
@@ -95,4 +95,31 @@ describe('App', () => {
...
+        reload: sinon.stub(),
+      },
+    };
+    const fakeCookieLib = {
+      remove: sinon.stub(),
+    };
+    const i18n = getFakeI18nInst();
+    const location = sinon.stub();
+    const root = renderIntoDocument(
+      <AppBase i18n={i18n}
```
Comment (tofumatt) [inline tests/client/amo/containers/TestApp.js]: Again, just ordering stuff but we have been putting `UpperCaseComponents` before `lowerCaseProps`.

File: src/amo/components/Footer.js (hunk 1 above)
Comment (muffinresearch) [inline src/amo/components/Footer.js]: Np - I wonder if we could lint for key sorting?

File: src/amo/containers/App.js (hunk 2 above)
Comment (muffinresearch) [inline src/amo/containers/App.js]: I was scared that was going to be a rickroll.

File: src/amo/components/Footer.js (hunk 1 above)
Comment (tofumatt) [inline src/amo/components/Footer.js]: At the very least I've filed #1387 so I don't forget.

File: tests/client/amo/containers/TestApp.js (hunk 3 above)
Comment (muffinresearch) [inline tests/client/amo/containers/TestApp.js]: potato

File: src/amo/components/Footer.js (hunk 1 above)
Comment (muffinresearch) [inline src/amo/components/Footer.js]: Good call, thanks!

**Pull request merged**